File: medplum-java/medplum-java-model/src/main/java/com/fhir/VerificationResult_PrimarySource.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableVerificationResult_PrimarySource.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface VerificationResult_PrimarySource {
  Optional<CodeableConcept> validationStatus();

  Optional<List<CodeableConcept>> type();

  Optional<Reference> who();

  Optional<dateTime> validationDate();

  Optional<List<CodeableConcept>> pushTypeAvailable();

  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<CodeableConcept> canPushUpdates();

  Optional<List<CodeableConcept>> communicationMethod();

  Optional<List<Extension>> modifierExtension();

  static ImmutableVerificationResult_PrimarySource.Builder builder() {
    return ImmutableVerificationResult_PrimarySource.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/VerificationResult_PrimarySource.ts
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { Reference } from "./Reference";

export interface VerificationResult_PrimarySource {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    who: Reference | null | undefined;
    type: CodeableConcept[] | null | undefined;
    communicationMethod: CodeableConcept[] | null | undefined;
    validationStatus: CodeableConcept | null | undefined;
    validationDate: dateTime | null | undefined;
    canPushUpdates: CodeableConcept | null | undefined;
    pushTypeAvailable: CodeableConcept[] | null | undefined;
}
